presto-blackhole/src/main/java/com/facebook/presto/plugin/blackhole/BlackHoleMetadata.java

```diff
@@ -0,0 +1,239 @@
+    {
+        //  returns null as the table does not contain sampled data
+        // (see {@link com.facebook.presto.spi.ConnectorMetadata.getSampleWeightColumnHandle()}
+        return null;
+    }
+
+    @Override
+    public boolean canCreateSampledTables(ConnectorSession session)
+    {
+        return false;
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorTableHandle tableHandle)
+    {
+        BlackHoleTableHandle blackHoleTableHandle = checkType(tableHandle, BlackHoleTableHandle.class, "tableHandle");
+        return blackHoleTableHandle.getColumnHandles().stream()
+                .collect(toMap(BlackHoleColumnHandle::getName, column -> column));
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        BlackHoleColumnHandle blackHoleColumnHandle = checkType(columnHandle, BlackHoleColumnHandle.class, "columnHandle");
+        return blackHoleColumnHandle.toColumnMetadata(typeManager);
+    }
+
+    @Override
+    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        return tables.values().stream()
+                .filter(table -> prefix.matches(table.toSchemaTableName()))
+                .collect(toMap(BlackHoleTableHandle::toSchemaTableName, handle -> handle.toTableMetadata(typeManager).getColumns()));
+    }
+
+    @Override
+    public void dropTable(ConnectorTableHandle tableHandle)
+    {
+        BlackHoleTableHandle blackHoleTableHandle = checkType(tableHandle, BlackHoleTableHandle.class, "tableHandle");
+        tables.remove(blackHoleTableHandle.getTableName());
+    }
+
+    @Override
+    public void renameTable(ConnectorTableHandle tableHandle, SchemaTableName newTableName)
+    {
+        BlackHoleTableHandle oldTableHandle = checkType(tableHandle, BlackHoleTableHandle.class, "tableHandle");
+        BlackHoleTableHandle newTableHandle = new BlackHoleTableHandle(
+                oldTableHandle.getSchemaName(),
+                newTableName.getTableName(),
+                oldTableHandle.getColumnHandles()
+        );
+        synchronized (tables) {
+            tables.remove(oldTableHandle.getTableName());
+            tables.put(newTableName.getTableName(), newTableHandle);
+        }
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        ConnectorOutputTableHandle outputTableHandle = beginCreateTable(session, tableMetadata);
+        commitCreateTable(outputTableHandle, ImmutableList.of());
+    }
+
+    @Override
+    public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        BlackHoleTableHandle blackHoleTableHandle = new BlackHoleTableHandle(tableMetadata);
+        tablesNotYetCommitted.put(tableMetadata.getTable().getTableName(), blackHoleTableHandle);
```

Comment:
Given that the `BlackholeTableHandle` contains all necessary information to create the table, it might be simpler to place it inside the `OutputTableHandle` to make the connector stateless with respect to keeping track of uncommitted tables. Otherwise, you have to deal with potential leaks (since there's no "abort" action) to clean up the entries in `tablesNotYetCommitted`